### DISCUSSION TITLE
Create Form items like radio buttons, push buttons

### DISCUSSION THREAD

**Discussion by sanksara** [pikepdf/pikepdf#562] [Q&A] [answered]
Requesting, how to create a form item like form push button, text box etc on a pdf page using pikepdf...
i have searched extensively and could only print existing form items using Annot function.
I would also need to create a one on a pdf page /document with no form elements

**Comment by jbarlow83** [comment]
You can create the data structures described in the PDF reference manual (linked in the documentation). There's no high level interface for this, unfortunately.

**Reply by sanksara** [reply]
Thanks for your quick reply,
i have been trying to create one from pdf structure. I am not very good in coding. could you help guide a sample code to create basic structure using pikepdf of annotation from which I can pickup

**Reply by jbarlow83** [reply, marked as answer]
The best way to do is inspect an existing PDF with a form in it. You could use the "Preflight" tool in Acrobat or the free iText RUPS app to see PDF internals; these tools are pretty essential for anyone who works with PDFs. When you see the objects that correspond to the form, usually most of what you need to do is create a pikepdf.Dictionary() object that replicates the same data structure.

You may find it easier to use the Python "borb" library, which does have some basic support for creating PDF forms. (However, it's license is AGPL if that is a concern to you.)
